src/main/java/com/softwire/todos/TodoCheckerApp.java

```diff
@@ -100,6 +100,8 @@ private Multimap<JiraIssueReference, CodeTodo> groupTodosByJiraIssue(List<CodeTo
             .map(JiraProject::getIssueIdPattern)
             .collect(Collectors.toList());
 
+        Map<String, JiraIssueReference> issueReferencesByKey = new HashMap<>();

```

Comment:
The jiraClient is doing some caching which ensures that the objects it returns have the same hash code if they are for the same issue. We need to do that here as well, otherwise each todo gets a JiraIssueReference with a different hashcode

Reply:
Do you also need to add equals() and hashcode() to the new JiraIssueReference class?

Reply:
We can do without them here for now - this could do with a bit of a refactor to make the behaviour more obvious though (but I was keen to merge this in to prevent incorrect comments from being written to JIRA on projects that use this)